h2o-algos/src/main/java/hex/ensemble/StackedEnsemble.java

```diff
@@ -57,6 +59,42 @@ public boolean isSupervised() {
     return true;
   }
 
+  @Override
+  protected void ignoreBadColumns(int npredictors, boolean expensive){
+    String msg = "Dropping bad and constant columns: ";
+
+    HashSet usedColumns = new HashSet();
+
+    for(Key k: _parms._base_models) {
+      Model model = (Model) DKV.getGet(k);
+      usedColumns.add(model._parms._response_column);
+      usedColumns.addAll(Arrays.asList(model._parms.getNonPredictors()));
+      if (model._output._origNames != null)
+        usedColumns.addAll(Arrays.asList(model._output._origNames));
+      else
+        usedColumns.addAll(Arrays.asList(model._output._names));
+    }
+
+    usedColumns.addAll(Arrays.asList(_parms.getNonPredictors()));
+
+    if (usedColumns.size() < _train._names.length) {
+      int[] toDelete = IntStream
+              .range(0, _train._names.length)
+              .filter(columnIdx -> !usedColumns.contains(_train._names[columnIdx]))
+              .toArray();
+
+      _removedCols = Arrays.stream(toDelete)
+              .mapToObj(columnIdx -> _train._names[columnIdx])
+              .collect(Collectors.toCollection(HashSet<String>::new));
```

Comment:
`Collectors.toSet()` should work

Reply:
Thanks!

Reply:
`ModelBuilder._removedCols` is declared as `HashSet` and AFAIK `Collectors.toSet()` doesn't guarantee that I will get a hash set so for now I decided to stick with this.

Reply:
I see. The declaration of `_removedCols` in `ModelBuilder` as `HashSet` instead of `Set` is most probably unjustified (as well as the `public` visibility) but well…

Now that you mention `_removedCols`, wouldn't it make sense to change lines 81 onwards to use `FilterCols` instead? This would avoid to logic copied from there.
```java
      new FilterCols(npredictors) {
        @Override protected boolean filter(Vec v, String name) {
             return !usedColumns.contains(name);
        }
      }.doIt(_train,"Dropping bad and constant columns: ",expensive);
```
This just requires to modify `FilterCols` to change the signature of `filter` method: given the implementation, it should be very simple to pass the `name` as well.

Reply:
Thanks for the suggestion!